### PR TITLE
Add Additional Configuration for OneSignal

### PR DESCRIPTION
**Pull request by richarddubay**
### What does this PR do, or why is it needed?
This PR includes more updates, particularly for iOS, to attempt to get push notifications with OneSignal working. 

Vincent suggested that I take a look at the OneSignal documentation, specifically starting with the "Add Required Capabilities" section (https://documentation.onesignal.com/docs/react-native-sdk-setup#section-add-required-capabilities) and go from there. So that's what I did. 

NOTE: I honestly have no idea if any of these changes will make a difference. 

### How do I test this PR?
Currently when you open the TestFlight build of the app it adds your device to OneSignal but there is an error. 
Screenshot 2019-05-09 18 51 33

These changes are an effort to mitigate that error.

---
> I am affirming this is my _best_ work ([Ecclesiastes 9:10](https://www.bible.com/bible/97/ECC.9.10.MSG))

## TODO

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] Part of [DEV-3620](https://newspring.atlassian.net/browse/DEV-3620)
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android if applicable
- [x] Set two relevant reviewers

## REVIEW

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

---

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Review by vinnyjth — DISMISSED** [review]
Talked with Rich about this over slack :) Posting my thoughts here for posterity. 

1. That push notification extension will break app center builds :confused: you don’t need it to have 99% of OneSignal features, so I would remove it

2. I’m pretty sure your issue is the missing entitlement, which you have taken care of in this PR :)

File: packages/newspringchurchapp/ios/newspringchurchapp/Info.plist [hunk 1]
```diff
@@ -86,6 +86,10 @@
 		<string>Colfax-Regular.otf</string>
 		<string>Colfax-RegularItalic.otf</string>
 	</array>
+	<key>UIBackgroundModes</key>
+	<array>
+		<string>remote-notification</string>
+	</array>
```
Comment (redreceipt) [inline packages/newspringchurchapp/ios/newspringchurchapp/Info.plist:92]: so what does this do?

Comment (richarddubay) [reply packages/newspringchurchapp/ios/newspringchurchapp/Info.plist:92]: I added it because it was called out in the OneSignal docs here (https://documentation.onesignal.com/docs/react-native-sdk-setup#section-add-required-capabilities). But for context it does this: https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/pushing_updates_to_your_app_silently

**Review by IsaacHardy — APPROVED** [review]
Hold on to your butts.
